### PR TITLE
feat(3000): Create new items from sidebar

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

There was no "New item" button in the 3000 sidebar.

## Changes

Now there is, for all supported item categories! There are three types of item creation:

1. Inline, for toolbar sites:
  inline

2. Modal, for dashboards and annotations:
  modal

3. Page, for all other categories (feature flags, insights, etc.)
  page

### CODE REVIEW

File: frontend/src/layout/navigation-3000/navigationLogic.ts [hunk 1]
```diff
@@ -117,8 +128,62 @@ export const navigation3000Logic = kea<navigation3000LogicType>([
...
     }),
     listeners(({ actions, values }) => ({
+        initiateNewItemInCategory: ({ category: categoryKey }) => {
+            const category = values.activeNavbarItem?.logic.values.contents?.find((item) => item.key === categoryKey)
+            if (!category) {
+                throw new Error(`Sidebar category '${categoryKey}' doesn't exist`)
+            } else if (!category.onAdd || typeof category.onAdd !== 'function') {
+                throw new Error(`Sidebar category '${categoryKey}' doesn't support onAdd`)
+            }
+            if (category.onAdd.length === 0) {
```
Comment (thmsobrmlr) [inline frontend/src/layout/navigation-3000/navigationLogic.ts:158]: I never knew you could use `.length` to check for number of arguments in JS - learned something new today.

File: frontend/src/layout/navigation-3000/navigationLogic.ts [hunk 2]
```diff
@@ -202,10 +267,39 @@ export const navigation3000Logic = kea<navigation3000LogicType>([
             },
         ],
         sidebarContentsFlattened: [
-            (s) => [(state) => s.activeNavbarItem(state)?.logic.findMounted()?.selectors.contents(state)],
```
Comment (thmsobrmlr) [inline frontend/src/layout/navigation-3000/navigationLogic.ts:205]: The following "silent" error in console seems likely related to this:

Screenshot 2023-07-18 at 10 24 42

**Review by thmsobrmlr — APPROVED** [review]
Overall looks very good to me. I'm leaving a couple of comments and you can decide yourself which ones are worth fixing here vs. in a follow up.

- Saving a new insight doesn't add it to the sidebar.
- In general it's a bit surprising that sometimes items wouldn't be on top e.g. experiments, maybe highlighting and scrolling to the item would help here, but I guess that's unncessary polish for now.
- Don't know if it's related to this PR, but switching to persons causes the following error for me:
  Screenshot 2023-07-18 at 10 22 34
  Screenshot 2023-07-18 at 10 27 59

File: frontend/src/layout/navigation-3000/navigationLogic.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/layout/navigation-3000/navigationLogic.ts:158]: You can 😄 #JustJSThings

File: frontend/src/scenes/urls.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ export const urls = {
         `/dashboard/${id}/subscriptions/${subscriptionId}`,
 
     sharedDashboard: (shareToken: string): string => `/shared_dashboard/${shareToken}`,
-    createAction: (): string => `/data-management/actions/new`,
+    createAction: (): string => `/data-management/actions/new`, // TODO: Convert all to simply .action('new')
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/scenes/urls.ts:39]: For some scenes we use `url.foo('new')` and only exports and actions have `url.createFoo()`. But maybe gonna remove this flyby comment for now, on second thought it's _fine_

File: frontend/src/layout/navigation-3000/navigationLogic.ts (hunk 2 above)
Comment (Twixes) [reply frontend/src/layout/navigation-3000/navigationLogic.ts:205]: Fixed

**Comment by Twixes** [comment]
Thanks for a thorough review @thmsobrmlr! Fixed all the issues, except the one with switching to persons, which I couldn't reproduce 🤔 
As for items not being on top, we definitely could use more consistent sorting order. Will take care of that in a follow-up PR.

**Pull request merged**